### PR TITLE
Test api

### PR DESCRIPTION
**Pull request by GevorgKhachatryan1997**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -33,10 +33,18 @@ android {
...
+    implementation 'androidx.core:core-ktx:1.8.0'
     implementation 'androidx.appcompat:appcompat:1.5.0'
     implementation 'com.google.android.material:material:1.6.1'
     implementation 'androidx.constraintlayout:constraintlayout:2.1.4'
+    //retrofit lib
+    implementation 'com.squareup.retrofit2:retrofit:2.9.0'
+    // viewmodelscope
+    implementation 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.5.1'
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.5.1'
     testImplementation 'junit:junit:4.13.2'
```
Comment (Artavazd-Khachatryan) [inline app/build.gradle:44]: add new line before test libraries

File: app/src/main/java/com/example/therundown/data/Repository.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.example.therundown.data
+
+import com.example.therundown.domain.NbaManager
+import com.example.therundown.domain.Player
+
+object Repository {
+
+    //TODO chek Response is success
```
Comment (Artavazd-Khachatryan) [inline app/src/main/java/com/example/therundown/data/Repository.kt]: fix this TODO

File: app/src/main/java/com/example/therundown/data/Repository.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    fun getPlayers(): List<Player>? {
```
Comment (Artavazd-Khachatryan) [inline app/src/main/java/com/example/therundown/data/Repository.kt]: maybe return type should be nonull after fixing TODO

File: app/src/main/java/com/example/therundown/domain/NbaManager.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+object NbaManager {
+
+    private val retrofit =
+        Retrofit
+            .Builder()
+            .baseUrl("https://free-nba.p.rapidapi.com/")
+            .addConverterFactory(GsonConverterFactory.create())
+            .build()
+
+    private val service = retrofit.create(NbaService::class.java)
```
Comment (Artavazd-Khachatryan) [inline app/src/main/java/com/example/therundown/domain/NbaManager.kt]: maybe we can do service creatin in one line

File: app/src/main/java/com/example/therundown/domain/NbaManager.kt [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.example.therundown.domain
+
+import com.example.therundown.data.Data
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+object NbaManager {
```
Comment (Artavazd-Khachatryan) [inline app/src/main/java/com/example/therundown/domain/NbaManager.kt]: maybe change name to NBAClient or NBAApi

File: app/src/main/java/com/example/therundown/domain/NbaManager.kt [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+    private val retrofit =
+        Retrofit
+            .Builder()
+            .baseUrl("https://free-nba.p.rapidapi.com/")
```
Comment (Artavazd-Khachatryan) [inline app/src/main/java/com/example/therundown/domain/NbaManager.kt]: make constant for base uri

File: app/src/main/java/com/example/therundown/domain/Player.kt [hunk 7]
```diff
@@ -0,0 +1,16 @@
+package com.example.therundown.domain
+
+import com.google.gson.annotations.SerializedName
+
+class Player(
```
Comment (Artavazd-Khachatryan) [inline app/src/main/java/com/example/therundown/domain/Player.kt]: add an empty line after fields

File: app/build.gradle [hunk 8]
```diff
@@ -33,10 +33,18 @@ android {
 
 dependencies {
```
Comment (Artavazd-Khachatryan) [inline app/build.gradle:34]: squash commits and changes the message to "Implement  NBA Api call"

**Review by Artavazd-Khachatryan — CHANGES_REQUESTED** [review]

File: app/build.gradle (hunk 1 above)
Comment (GevorgKhachatryan1997) [reply app/build.gradle:44]: fixed

File: app/src/main/java/com/example/therundown/domain/NbaManager.kt (hunk 4 above)
Comment (GevorgKhachatryan1997) [reply app/src/main/java/com/example/therundown/domain/NbaManager.kt]: fixed

File: app/src/main/java/com/example/therundown/domain/NbaManager.kt (hunk 5 above)
Comment (GevorgKhachatryan1997) [reply app/src/main/java/com/example/therundown/domain/NbaManager.kt]: fixed

File: app/src/main/java/com/example/therundown/domain/NbaManager.kt (hunk 6 above)
Comment (GevorgKhachatryan1997) [reply app/src/main/java/com/example/therundown/domain/NbaManager.kt]: fixed

File: app/src/main/java/com/example/therundown/domain/Player.kt (hunk 7 above)
Comment (GevorgKhachatryan1997) [reply app/src/main/java/com/example/therundown/domain/Player.kt]: fixed

File: app/src/main/java/com/example/therundown/data/Repository.kt (hunk 3 above)
Comment (GevorgKhachatryan1997) [reply app/src/main/java/com/example/therundown/data/Repository.kt]: fixed

File: app/src/main/java/com/example/therundown/data/Repository.kt (hunk 2 above)
Comment (GevorgKhachatryan1997) [reply app/src/main/java/com/example/therundown/data/Repository.kt]: fixed

**Pull request merged**